bleps/src/lib.rs

```diff
@@ -594,4 +594,54 @@ pub mod asynch {
             data
         }
     }
+
+    pub trait AsyncHciConnection {
+        async fn read(&self) -> Option<u8>;
+    
+        async fn write(&self, data: u8);
+    
+        fn millis(&self) -> u64;
+    }
+    
+    pub struct AsyncHciConnector<T>
+    where
+        T: embedded_io_async::Read + embedded_io_async::Write,
+    {
+        hci: RefCell<T>,
+        get_millis: fn() -> u64,
+    }
+    
+    impl<T> AsyncHciConnector<T>
+    where
+        T: embedded_io_async::Read + embedded_io_async::Write,
+    {
+        pub fn new(hci: T, get_millis: fn() -> u64) -> AsyncHciConnector<T> {
+            AsyncHciConnector {
+                hci: RefCell::new(hci),
+                get_millis,
+            }
+        }
+    }
+    
+    impl<T> AsyncHciConnection for AsyncHciConnector<T>
+    where
+        T: embedded_io_async::Read + embedded_io_async::Write,
+    {
+        async fn read(&self) -> Option<u8> {
```

Comment:
1. should be reading/writing 1 byte at a time
2. should we be returning `Result<usize, SomeKindOfAsyncErrorWrapperThingie>`?